core/src/HIP/Kokkos_HIP_IsXnack.hpp

```diff
@@ -0,0 +1,93 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+/*--------------------------------------------------------------------------*/
+
+#ifndef KOKKOS_HIP_ISXNACK_HPP
+#define KOKKOS_HIP_ISXNACK_HPP
+
+#include <string>
+#include <optional>
+#include <fstream>
+
+#ifdef __linux__
+#include <stdio.h>
+#include <sys/utsname.h>
+#endif
+
+namespace Kokkos::Impl {
+
+inline bool is_hsa_xnack_1_impl() {
+  const char* envVar = std::getenv("HSA_XNACK");
+  return envVar != nullptr && std::string(envVar) == "1";
+}
+
+// returns true iff environment variable HSA_XNACK=1
+inline bool is_hsa_xnack_1() {
+  static bool cache = is_hsa_xnack_1_impl();
```

Comment:
Prefer an immediately involved lambda and a string view
```suggestion
  static bool cache = [] {
    const char* var = std::getenv("HSA_XNACK");
    return var && std::string_view{var} == "1";
  }();
```

You'd also need to include `<string_view>` instead of `<string>`